src/LinkDotNet.NCronJob/Scheduler/CronScheduler.cs

```diff
@@ -119,11 +145,16 @@ private void ScheduleInitialJobs()
 
     private void ScheduleJob(RegistryEntry job)
     {
-        var utcNow = timeProvider.GetUtcNow().DateTime;
-        var nextRunTime = job.CrontabSchedule?.GetNextOccurrence(utcNow) ?? DateTime.MaxValue;
-        // higher means more priority
-        var priorityValue = (int)job.Priority;
-        jobQueue.Enqueue(job, (nextRunTime, priorityValue));
+        var utcNow = timeProvider.GetUtcNow();
+        var nextRunTime = job.CronExpression!.GetNextOccurrence(utcNow, job.TimeZone);
+
+        if (nextRunTime.HasValue)
```

Comment:
If you have a valid CRON expression - isn't that guaranteed?

I know that Quartz has some crazy stuff where you explicitly can set a year and whatnot, but with "standard"<sup>In the sense of Unix</sup> CRON you should always get a value. Did I miss something?

Reply:
No you're correct, but `GetNextOccurrence(utcNow, job.TimeZone)` returns nullable so just wrote it like that. Could've used null-forgiving operator but it just seemed visually more intuitive to write it like this.

Reply:
Ah no that is a good check tbh.
Shamelessly stolen from the docs of Cronos:

> The `nextUtc` will contain the next occurrence in UTC, *after the given time*, or `null` value when it is unreachable (for example, Feb 30).